tutorials/physics/rigid_body.rst

```diff
@@ -0,0 +1,51 @@
+.. _doc_rigid_body
+
+Rigid Body
+=========
+
+What is a rigid body?
+--------------------
+
+Rigid body is a rigid (non deformable) physical object that is directly controlled by physics engine in order to simulate phisical behaviour of objects.
+It's necessary to add one or more shapes that define the final shape of rigid body, you can set different position of these primitive shapes and this affect the COM (center of mass) of body.
+
+How to control rigid body
+--------------------------
+
+To alter the rigid body behaviour you can set some parameters like damping, friction, mass, bounce, gravity scale, etc..
+
+While to control the movement of the rigid body, you have many ways and you have to choose the correct one depending of the case.
+
+When you need to move a rigid body one time you can use all functions defined by *spatial node*, for example ``set_global_transform`` or ``look_at``. These functions are perfectly fine to create a teleport but must not be called each frame to move the rigid body because in this case the physics engine will not be able to handle it correctly and the physics simulation will not be correct.
+For example one common mistake is to use ``look_at`` each frame, this break physics similation. Few line below I'll show how to implement ``look_at`` correctly.
+
+The fact that you can't use ``set_global_transform`` or ``look_at`` functions doesn't mean that you can't have full control of rigid body, rather by adding forces, impulse, or by setting velocity, in the ``_integrate_forces(state)`` you will be able to achieve any movement you desire, even if it's not physically possible!
+
+Look at function
+----------------
+
+One of the most required feature is the look_at function, this function is defined by spatial node and as I said before it can't be used in each frame to follow a target with sight.
+Even if this function *seems to work* on the old physics engine use it is not correct and is not intended to work. So here I've prepared a custom ``look_at`` that will work with rigid body.
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    extends RigidBody
+
+    func look_follow(state, current_transform, target_location):
+	var up_dir = Vector3(0,1,0);
+	var cur_dir = current_transform.basis.xform(Vector3(0, 0, 1))
+	var target_dir = (target_location-current_transform.origin).normalized()
+	var rotation_angle = acos(cur_dir.x) - acos(target_dir.x)
+	
+	state.set_angular_velocity(up_dir * (rotation_angle / state.get_step()))
+
+    func _integrate_forces(state):
+	
+	var target = $"../Bullets5/MonitoringArea".get_global_transform().origin
+	look_follow(state, get_global_transform(), target)
+
+
+As you can see this function ``look_follow`` internally use the function ``set_angular_velocity`` to rotate the rigid body, it basically take the difference of current orientation and target position and add the velocity required to rotate the rigid body within 1 frame.
```

Comment:
This function uses the rigid body's ``set_angular_velocity()``method to rotate the body. It first calculates the difference between the current and desired angle and then adds the velocity needed to rotate by that amount in one frame's time.